### PR TITLE
[wasi] WasmTestRunnerDir

### PR DESCRIPTION
**Pull request by pavelsavara**
- fix `WasmTestRunnerDir`
- minor native build fixes

### CODE REVIEW

File: eng/testing/tests.wasi.targets [hunk 1]
```diff
@@ -29,7 +29,7 @@
 
     <_AppArgs Condition="'$(IsFunctionalTest)' != 'true' and '$(WasmSingleFileBundle)' != 'true'">$(_AppArgs) managed/$(AssemblyName).dll</_AppArgs>
     <_AppArgs Condition="'$(IsFunctionalTest)' != 'true' and '$(WasmSingleFileBundle)' == 'true'">$(_AppArgs) $(AssemblyName).dll</_AppArgs>
-    <_AppArgs Condition="'$(WasmTestAppArgs)' != ''">$(_AppArgs) -- $(WasmTestAppArgs)</_AppArgs>
```
Comment (ilonatommy) [inline eng/testing/tests.wasi.targets:32]: Why do we stop separating them?

**Review by maraf — APPROVED** [review]

File: eng/testing/tests.wasi.targets [hunk 2]
```diff
@@ -49,6 +49,8 @@
     <_XHarnessArgs Condition="'$(WasmXHarnessTestsTimeout)' != ''" >$(_XHarnessArgs) &quot;--timeout=$(WasmXHarnessTestsTimeout)&quot;</_XHarnessArgs>
     <_XHarnessArgs                                                 >$(_XHarnessArgs) --engine-arg=--wasm --engine-arg=max-wasm-stack=134217728</_XHarnessArgs>
     <_XHarnessArgs                                                 >$(_XHarnessArgs) --engine-arg=--wasi --engine-arg=http</_XHarnessArgs>
+    <_XHarnessArgs                                                 >$(_XHarnessArgs) --engine-arg=--wasi --engine-arg=inherit-network</_XHarnessArgs>
```
Comment (ilonatommy) [inline eng/testing/tests.wasi.targets:52]: Just for the future readers:
```
inherit-network[=y|n] -- Flag for WASI preview2 to inherit the host's network within the guest so it has full access to all addresses/ports/etc.
allow-ip-name-lookup[=y|n] -- Indicates whether `wasi:sockets/ip-name-lookup` is enabled or not.
```

**Review by ilonatommy — APPROVED** [review]

File: eng/testing/tests.wasi.targets (hunk 1 above)
Comment (pavelsavara) [reply eng/testing/tests.wasi.targets:32]: it didn't work as is, we just unified it with browser here

File: eng/testing/tests.wasi.targets (hunk 2 above)
Comment (pavelsavara) [reply eng/testing/tests.wasi.targets:52]: `wasmtime run -S help`

**Pull request merged**